### PR TITLE
feat: track scene load times

### PR DESCRIPTION
**Pull request by eordano**

### CODE REVIEW

File: kernel/packages/shared/loading/sagas.ts [hunk 1]
```diff
@@ -10,6 +14,32 @@ export const DELAY_BETWEEN_MESSAGES = 10 * SECONDS
...
   yield takeLatest(TELEPORT_TRIGGERED, changeSubtext)
+
+  yield takeEvery(SCENE_LOAD, trackLoadTime)
+}
+
+export function* trackLoadTime(action: SceneLoadAction): any {
+  const start = new Date().getTime()
+  const sceneId = action.payload
+  const result = yield race({
+    load: call(function*() {
```
Comment (moliva) [inline kernel/packages/shared/loading/sagas.ts]: Wouldn't it be simpler by inlining the `take((action: AnyAction) => action.type === SCENE_START && (action as SceneStartAction).payload === sceneId)` here?

File: kernel/packages/shared/loading/sagas.ts [hunk 2]
```diff
@@ -10,6 +14,32 @@ export const DELAY_BETWEEN_MESSAGES = 10 * SECONDS
...
+      yield take((action: AnyAction) => action.type === SCENE_START && (action as SceneStartAction).payload === sceneId)
+      return true
+    }),
+    fail: call(function*() {
+      yield take((action: AnyAction) => action.type === SCENE_FAIL && (action as SceneFailAction).payload === sceneId)
```
Comment (moliva) [inline kernel/packages/shared/loading/sagas.ts]: Same as above.

**Review by moliva — DISMISSED** [review]

**Review by moliva — APPROVED** [review]

File: kernel/packages/shared/loading/actions.ts [hunk 3]
```diff
@@ -8,6 +8,10 @@ export const signalSceneLoad = (sceneId: string) => action(SCENE_LOAD, sceneId)
 export const signalSceneStart = (sceneId: string) => action(SCENE_START, sceneId)
 export const signalSceneFail = (sceneId: string) => action(SCENE_FAIL, sceneId)
 
+export type SceneLoadAction = ReturnType<typeof signalSceneLoad>
+export type SceneStartAction = ReturnType<typeof signalSceneStart>
```
Comment (eordano) [inline kernel/packages/shared/loading/actions.ts]:
```suggestion

```

File: kernel/packages/shared/loading/actions.ts [hunk 4]
```diff
@@ -8,6 +8,10 @@ export const signalSceneLoad = (sceneId: string) => action(SCENE_LOAD, sceneId)
...
+export type SceneFailAction = ReturnType<typeof signalSceneFail>
```
Comment (eordano) [inline kernel/packages/shared/loading/actions.ts]:
```suggestion

```

**Review by pablitar — APPROVED** [review]
LGTM!

**Pull request merged**